external/external.go

```diff
@@ -115,15 +88,19 @@ func (e *External) CreateService(writer http.ResponseWriter, request *http.Reque
 
 	svc, err := e.ServiceController.Create(opts)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+		return err
 	}
 
 	log.Printf("CreateService: finished creating service %s", service)
 
-	WriteService(svc, writer)
+	returnOpts := &ServiceOptions{
+		Name:       svc.Name,
+		Namespace:  svc.Namespace,
+		ListenPort: svc.Spec.Ports[0].Port,
+		TargetPort: svc.Spec.Ports[0].TargetPort.IntValue(),
+	}
```

Comment:
I'd be a little tempted to create some conversion functions for common tasks such as these.